### DISCUSSION TITLE
Provide a specification for dfx.json

### DISCUSSION THREAD

**Discussion by nmattia** [dfinity/sdk#2649] [Feature Requests]
Was: https://github.com/dfinity/sdk/issues/1963

It would be great to have a specification for `dfx.json`. There is a [schema][dfx-json] available, but I am not sure how to read it (and is a bit hard to navigate).

Here are some examples of good references for config files:

* [npm's `package.json`][package-json]
* [cargo's `Cargo.toml`][cargo-toml]
* [GitHub Actions workflows][github-actions]

[package-json]: https://docs.npmjs.com/cli/v8/configuring-npm/package-json
[cargo-toml]: https://doc.rust-lang.org/cargo/reference/manifest.html
[github-actions]: https://docs.github.com/en/enterprise-server@3.3/actions/using-workflows/workflow-syntax-for-github-actions
[dfx-json]: https://internetcomputer.org/docs/current/references/dfx-json-reference/

**Comment by viviveevee** [comment]
This is already an internal ticket: [SDK-699](https://dfinity.atlassian.net/browse/SDK-699)
